channeld/full_channel.c

```diff
@@ -351,6 +351,11 @@ static enum channel_add_err add_htlc(struct channel *channel,
 	if (htlc->msatoshi < htlc_minimum_msat(channel, recipient)) {
 		return CHANNEL_ERR_HTLC_BELOW_MINIMUM;
 	}
+	if (htlc->msatoshi > htlc_maximum_msat(channel->funding_msat,
```

Comment:
This is definitely a good thought; in fact, there's a FIXME and an issue about this (#546).  We don't check that we're not trying to send more than their max_htlc_value_in_flight_msat, or exceeding their max_accepted_htlcs, which we absolutely should do.  Those checks would be a great PR!

But that's orthogonal to this; we already check there's no reserve violation in this code, we should NOT check for additional things.  For example, we *can* send more than max-2*reserve for our very first HTLC.